app/src/main/java/com/github/mobile/ui/comment/CommentListAdapter.java

```diff
@@ -101,31 +109,46 @@ protected void update(int position, final Comment comment) {
         setText(1, comment.getUser().getLogin());
         setText(2, TimeUtils.getRelativeTime(comment.getUpdatedAt()));
 
-        // Edit Comment ImageButton
-        final ImageView ivEdit = view(4);
-        if (editCommentListener != null) {
-            ivEdit.setOnClickListener(new OnClickListener() {
-                @Override
-                public void onClick(View v) {
-                    editCommentListener.onEditComment(comment);
-                }
-            });
-        } else {
-            ivEdit.setVisibility(View.GONE);
-        }
-
-        // Delete Comment ImageButton
-        final ImageView ivDelete = view(5);
-        if (deleteCommentListener != null) {
-            ivDelete.setOnClickListener(new OnClickListener() {
-                @Override
-                public void onClick(View v) {
-                    deleteCommentListener.onDeleteComment(comment);
+
+        final ImageView ivMore = view(4);
+        ivMore.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                showMorePopup(ivMore, comment);
+            }
+        });
+    }
+
+    private void showMorePopup(View v, final Comment comment) {
+        PopupMenu menu = new PopupMenu(context, v);
+        menu.inflate(R.menu.comment_popup);
+
+        boolean canEdit = editCommentListener != null;
+        boolean canDelete = deleteCommentListener != null;
+
+        menu.getMenu().findItem(R.id.m_edit).setEnabled(canEdit);
+        menu.getMenu().findItem(R.id.m_delete).setEnabled(canDelete);
```

Comment:
Hmm, @fadils is checking whether or not a `editCommentListener` or `deleteCommentListener` is provided the proper way to check? That was what I gathered from looking at the previous code, which would hide those views entirely.